server/src/main/java/org/opensearch/wlm/tracker/TaskResourceUsageCalculator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.tracker;
+
+import org.opensearch.core.tasks.resourcetracker.ResourceStats;
+import org.opensearch.tasks.Task;
+import org.opensearch.wlm.ResourceType;
+
+import java.util.function.Supplier;
+
+import static org.opensearch.wlm.tracker.QueryGroupResourceUsageTrackerService.HEAP_SIZE_BYTES;
+
+/**
+ * Utility class to calculate task level resource usage
+ */
+public abstract class TaskResourceUsageCalculator {
+    public static TaskResourceUsageCalculator from(final ResourceType resourceType) {
+        if (resourceType == ResourceType.CPU) {
+            return new TaskCpuUsageCalculator();
+        } else if (resourceType == ResourceType.MEMORY) {
+            return new TaskMemoryUsageCalculator();
+        }
```

Comment:
These classes should be singleton, else we are adding unnecessary GC overhead

Reply:
Removed the class